ui/pages/send/send-content/send-content.component.js

```diff
@@ -84,24 +88,52 @@ export default class SendContent extends Component {
     );
   }
 
+  checkContractAddress = () => {
+    const { recipient, tokenAddressList } = this.props;
+
+    getSymbolAndDecimals(recipient.userInput, tokenAddressList).then(
+      (result) => {
+        if (
+          (result.symbol !== undefined && result.symbol !== '') ||
+          (result.decimals !== undefined && result.decimals !== '')
+        ) {
+          this.setState({ isContractAddress: true });
+        }
+      },
+    );
+  };
+
   maybeRenderAddContact() {
     const { t } = this.context;
-    const { isOwnedAccount, contact = {}, to } = this.props;
-    const { showNicknamePopovers } = this.state;
+    const { isOwnedAccount, contact = {}, to, recipient } = this.props;
+    const { showNicknamePopovers, isContractAddress } = this.state;
+
+    this.checkContractAddress();
 
     if (isOwnedAccount || contact.name) {
-      return null;
+      return recipient.warning === 'knownAddressRecipient' ||
+        isContractAddress ? (
+        <Dialog type="warning" className="send__error-dialog">
+          {t('knownAddressRecipient')}
+        </Dialog>
+      ) : null;
     }
 
     return (
       <>
+        {recipient.warning === 'knownAddressRecipient' || isContractAddress ? (
+          <Dialog type="warning" className="send__error-dialog">
+            {t('knownAddressRecipient')}
```

Comment:
Wondering if we should rename this key to `knownContractAddressWarning`? cc @danjm

Reply:
Did not change anything. Waiting for Dan's response on this one.

Reply:
Let's not change it right now, because then we have to update all of the translations too